connector/src/main/java/org/jboss/as/connector/services/resourceadapters/deployment/ResourceAdapterXmlDeploymentService.java

```diff
@@ -214,17 +214,14 @@ protected org.jboss.jca.core.spi.security.SubjectFactory getSubjectFactory(
             if (securityMetadata == null)
                 return null;
             final String securityDomain = securityMetadata.resolveSecurityDomain();
-            if (securityMetadata instanceof org.jboss.as.connector.metadata.api.common.SecurityMetadata &&
-                    ((org.jboss.as.connector.metadata.api.common.SecurityMetadata)securityMetadata).isElytronEnabled()) {
-                try {
-                    return new ElytronSubjectFactory(null, new URI(jndiName));
-                } catch (URISyntaxException e) {
-                    throw ConnectorLogger.ROOT_LOGGER.cannotDeploy(e);
-                }
-            } else if (securityDomain == null || securityDomain.trim().equals("")) {
-                return null;
+            if (securityDomain != null) {
+                throw new DeployException(ConnectorLogger.SUBSYSTEM_RA_LOGGER.legacySecurityNotSupported());
```

Comment:
@tadamski I suspect this causes test failures because securityMetadata.resolveSecurityDomain() is returning the name of the authentication context (e.g. RaOperationUtil passes that into CredentialImpl.)

Reply:
fixed